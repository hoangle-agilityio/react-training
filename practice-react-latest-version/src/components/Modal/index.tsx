// Libs
import {
  Modal as ReactModal,
  ModalCloseButton,
  ModalContent,
  ModalHeader,
  ModalOverlay,
  ModalContentProps,
  ModalHeaderProps,
} from "@chakra-ui/react";
import React, { memo } from "react";

type StyleTypes = {
  wrapper?: ModalContentProps;
  title?: ModalHeaderProps;
};

type Props = {
  title: string;
  children: React.ReactNode;
  isOpen: boolean;
  size?: "xs" | "sm" | "md" | "lg" | "xl" | "full";
  styles?: StyleTypes;
  onClose: () => void;
};

const Modal = ({
  title,
  children,
  isOpen,
  size = "md",
  styles,
  onClose,
}: Props) => {
  return (
    <ReactModal isOpen={isOpen} onClose={onClose} size={size}>
      <ModalOverlay>
        <ModalContent {...styles?.wrapper} padding="20px 35px">
          <ModalCloseButton />
          <ModalHeader {...styles?.title} padding="0">
            {title}
          </ModalHeader>
          {children}
        </ModalContent>
      </ModalOverlay>
    </ReactModal>
  );
};

export default memo(Modal);
