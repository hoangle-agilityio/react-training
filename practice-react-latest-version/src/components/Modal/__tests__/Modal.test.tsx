// Libs
import { render } from "@testing-library/react";
import * as React from "react";
import { Text } from "@chakra-ui/react";

// Components
import Modal from "..";

const children: React.ReactNode = (
  <Text fontSize="base" margin="20px 0" data-testid="modal-content">
    Lorem Ipsum is simply dummy text of the printing and typesetting industry.
  </Text>
);

const props = {
  title: "Add New Customer",
  onClose: jest.fn(),
  isOpen: true,
  children,
};

describe("Modal Component", () => {
  test("Should match snapshoot for Modal", () => {
    const dom = render(<Modal {...props} />);

    expect(dom).toMatchSnapshot();
  });

  test("Should match snapshoot for Modal with size xs", () => {
    const dom = render(<Modal {...props} size="xs" />);

    expect(dom).toMatchSnapshot();
  });

  test("Should match snapshoot for Modal with size md", () => {
    const dom = render(<Modal {...props} size="md" />);

    expect(dom).toMatchSnapshot();
  });

  test("Should match snapshoot for Modal with size sm", () => {
    const dom = render(<Modal {...props} size="sm" />);

    expect(dom).toMatchSnapshot();
  });

  test("Should match snapshoot for Modal with size lg", () => {
    const dom = render(<Modal {...props} size="lg" />);

    expect(dom).toMatchSnapshot();
  });

  test("Should match snapshoot for Modal with size xl", () => {
    const dom = render(<Modal {...props} size="xl" />);

    expect(dom).toMatchSnapshot();
  });

  test("Should match snapshoot for Modal with size full", () => {
    const dom = render(<Modal {...props} size="full" />);

    expect(dom).toMatchSnapshot();
  });

  test("Doesn't render modal if prop isOpen is false", () => {
    const dom = render(<Modal {...props} isOpen={false} />);

    expect(dom.findAllByTestId("modal-content")).toMatchObject({});
  });
});
