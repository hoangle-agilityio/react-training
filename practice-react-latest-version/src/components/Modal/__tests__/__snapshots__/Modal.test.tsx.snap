// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Component Should match snapshoot for Modal 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div
      aria-hidden="true"
      data-aria-hidden="true"
    />
    <div
      class="chakra-portal"
    >
      <div
        class="chakra-modal__overlay css-13d1tkw"
        style="opacity: 0;"
      >
        <div
          aria-hidden="true"
          data-aria-hidden="true"
          data-focus-guard="true"
          style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
          tabindex="0"
        />
        <div
          data-focus-lock-disabled="false"
        >
          <div
            class="chakra-modal__content-container css-1oxhx2p"
            tabindex="-1"
          >
            <section
              aria-labelledby="chakra-modal--header-:r0:"
              aria-modal="true"
              class="chakra-modal__content css-1bn8s9p"
              id="chakra-modal-:r0:"
              role="dialog"
              style="opacity: 0; transform: scale(0.95) translateZ(0);"
              tabindex="-1"
            >
              <button
                aria-label="Close"
                class="chakra-modal__close-btn css-6sdaxy"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="chakra-icon css-onkibi"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M.439,21.44a1.5,1.5,0,0,0,2.122,2.121L11.823,14.3a.25.25,0,0,1,.354,0l9.262,9.263a1.5,1.5,0,1,0,2.122-2.121L14.3,12.177a.25.25,0,0,1,0-.354l9.263-9.262A1.5,1.5,0,0,0,21.439.44L12.177,9.7a.25.25,0,0,1-.354,0L2.561.44A1.5,1.5,0,0,0,.439,2.561L9.7,11.823a.25.25,0,0,1,0,.354Z"
                    fill="currentColor"
                  />
                </svg>
              </button>
              <header
                class="chakra-modal__header css-epw6ly"
                id="chakra-modal--header-:r0:"
              >
                Add New Customer
              </header>
              <p
                class="chakra-text css-bp306l"
                data-testid="modal-content"
              >
                Lorem Ipsum is simply dummy text of the printing and typesetting industry.
              </p>
            </section>
          </div>
        </div>
        <div
          aria-hidden="true"
          data-aria-hidden="true"
          data-focus-guard="true"
          style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
          tabindex="0"
        />
      </div>
    </div>
  </body>,
  "container": <div
    aria-hidden="true"
    data-aria-hidden="true"
  />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Modal Component Should match snapshoot for Modal with size full 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div
      aria-hidden="true"
      data-aria-hidden="true"
    />
    <div
      class="chakra-portal"
    >
      <div
        class="chakra-modal__overlay css-13d1tkw"
        style="opacity: 0;"
      >
        <div
          aria-hidden="true"
          data-aria-hidden="true"
          data-focus-guard="true"
          style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
          tabindex="0"
        />
        <div
          data-focus-lock-disabled="false"
        >
          <div
            class="chakra-modal__content-container css-1oxhx2p"
            tabindex="-1"
          >
            <section
              aria-labelledby="chakra-modal--header-:r6:"
              aria-modal="true"
              class="chakra-modal__content css-1bn8s9p"
              id="chakra-modal-:r6:"
              role="dialog"
              style="opacity: 0; transform: scale(0.95) translateZ(0);"
              tabindex="-1"
            >
              <button
                aria-label="Close"
                class="chakra-modal__close-btn css-6sdaxy"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="chakra-icon css-onkibi"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M.439,21.44a1.5,1.5,0,0,0,2.122,2.121L11.823,14.3a.25.25,0,0,1,.354,0l9.262,9.263a1.5,1.5,0,1,0,2.122-2.121L14.3,12.177a.25.25,0,0,1,0-.354l9.263-9.262A1.5,1.5,0,0,0,21.439.44L12.177,9.7a.25.25,0,0,1-.354,0L2.561.44A1.5,1.5,0,0,0,.439,2.561L9.7,11.823a.25.25,0,0,1,0,.354Z"
                    fill="currentColor"
                  />
                </svg>
              </button>
              <header
                class="chakra-modal__header css-epw6ly"
                id="chakra-modal--header-:r6:"
              >
                Add New Customer
              </header>
              <p
                class="chakra-text css-bp306l"
                data-testid="modal-content"
              >
                Lorem Ipsum is simply dummy text of the printing and typesetting industry.
              </p>
            </section>
          </div>
        </div>
        <div
          aria-hidden="true"
          data-aria-hidden="true"
          data-focus-guard="true"
          style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
          tabindex="0"
        />
      </div>
    </div>
  </body>,
  "container": <div
    aria-hidden="true"
    data-aria-hidden="true"
  />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Modal Component Should match snapshoot for Modal with size lg 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div
      aria-hidden="true"
      data-aria-hidden="true"
    />
    <div
      class="chakra-portal"
    >
      <div
        class="chakra-modal__overlay css-13d1tkw"
        style="opacity: 0;"
      >
        <div
          aria-hidden="true"
          data-aria-hidden="true"
          data-focus-guard="true"
          style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
          tabindex="0"
        />
        <div
          data-focus-lock-disabled="false"
        >
          <div
            class="chakra-modal__content-container css-1oxhx2p"
            tabindex="-1"
          >
            <section
              aria-labelledby="chakra-modal--header-:r4:"
              aria-modal="true"
              class="chakra-modal__content css-1bn8s9p"
              id="chakra-modal-:r4:"
              role="dialog"
              style="opacity: 0; transform: scale(0.95) translateZ(0);"
              tabindex="-1"
            >
              <button
                aria-label="Close"
                class="chakra-modal__close-btn css-6sdaxy"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="chakra-icon css-onkibi"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M.439,21.44a1.5,1.5,0,0,0,2.122,2.121L11.823,14.3a.25.25,0,0,1,.354,0l9.262,9.263a1.5,1.5,0,1,0,2.122-2.121L14.3,12.177a.25.25,0,0,1,0-.354l9.263-9.262A1.5,1.5,0,0,0,21.439.44L12.177,9.7a.25.25,0,0,1-.354,0L2.561.44A1.5,1.5,0,0,0,.439,2.561L9.7,11.823a.25.25,0,0,1,0,.354Z"
                    fill="currentColor"
                  />
                </svg>
              </button>
              <header
                class="chakra-modal__header css-epw6ly"
                id="chakra-modal--header-:r4:"
              >
                Add New Customer
              </header>
              <p
                class="chakra-text css-bp306l"
                data-testid="modal-content"
              >
                Lorem Ipsum is simply dummy text of the printing and typesetting industry.
              </p>
            </section>
          </div>
        </div>
        <div
          aria-hidden="true"
          data-aria-hidden="true"
          data-focus-guard="true"
          style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
          tabindex="0"
        />
      </div>
    </div>
  </body>,
  "container": <div
    aria-hidden="true"
    data-aria-hidden="true"
  />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Modal Component Should match snapshoot for Modal with size md 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div
      aria-hidden="true"
      data-aria-hidden="true"
    />
    <div
      class="chakra-portal"
    >
      <div
        class="chakra-modal__overlay css-13d1tkw"
        style="opacity: 0;"
      >
        <div
          aria-hidden="true"
          data-aria-hidden="true"
          data-focus-guard="true"
          style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
          tabindex="0"
        />
        <div
          data-focus-lock-disabled="false"
        >
          <div
            class="chakra-modal__content-container css-1oxhx2p"
            tabindex="-1"
          >
            <section
              aria-labelledby="chakra-modal--header-:r2:"
              aria-modal="true"
              class="chakra-modal__content css-1bn8s9p"
              id="chakra-modal-:r2:"
              role="dialog"
              style="opacity: 0; transform: scale(0.95) translateZ(0);"
              tabindex="-1"
            >
              <button
                aria-label="Close"
                class="chakra-modal__close-btn css-6sdaxy"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="chakra-icon css-onkibi"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M.439,21.44a1.5,1.5,0,0,0,2.122,2.121L11.823,14.3a.25.25,0,0,1,.354,0l9.262,9.263a1.5,1.5,0,1,0,2.122-2.121L14.3,12.177a.25.25,0,0,1,0-.354l9.263-9.262A1.5,1.5,0,0,0,21.439.44L12.177,9.7a.25.25,0,0,1-.354,0L2.561.44A1.5,1.5,0,0,0,.439,2.561L9.7,11.823a.25.25,0,0,1,0,.354Z"
                    fill="currentColor"
                  />
                </svg>
              </button>
              <header
                class="chakra-modal__header css-epw6ly"
                id="chakra-modal--header-:r2:"
              >
                Add New Customer
              </header>
              <p
                class="chakra-text css-bp306l"
                data-testid="modal-content"
              >
                Lorem Ipsum is simply dummy text of the printing and typesetting industry.
              </p>
            </section>
          </div>
        </div>
        <div
          aria-hidden="true"
          data-aria-hidden="true"
          data-focus-guard="true"
          style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
          tabindex="0"
        />
      </div>
    </div>
  </body>,
  "container": <div
    aria-hidden="true"
    data-aria-hidden="true"
  />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Modal Component Should match snapshoot for Modal with size sm 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div
      aria-hidden="true"
      data-aria-hidden="true"
    />
    <div
      class="chakra-portal"
    >
      <div
        class="chakra-modal__overlay css-13d1tkw"
        style="opacity: 0;"
      >
        <div
          aria-hidden="true"
          data-aria-hidden="true"
          data-focus-guard="true"
          style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
          tabindex="0"
        />
        <div
          data-focus-lock-disabled="false"
        >
          <div
            class="chakra-modal__content-container css-1oxhx2p"
            tabindex="-1"
          >
            <section
              aria-labelledby="chakra-modal--header-:r3:"
              aria-modal="true"
              class="chakra-modal__content css-1bn8s9p"
              id="chakra-modal-:r3:"
              role="dialog"
              style="opacity: 0; transform: scale(0.95) translateZ(0);"
              tabindex="-1"
            >
              <button
                aria-label="Close"
                class="chakra-modal__close-btn css-6sdaxy"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="chakra-icon css-onkibi"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M.439,21.44a1.5,1.5,0,0,0,2.122,2.121L11.823,14.3a.25.25,0,0,1,.354,0l9.262,9.263a1.5,1.5,0,1,0,2.122-2.121L14.3,12.177a.25.25,0,0,1,0-.354l9.263-9.262A1.5,1.5,0,0,0,21.439.44L12.177,9.7a.25.25,0,0,1-.354,0L2.561.44A1.5,1.5,0,0,0,.439,2.561L9.7,11.823a.25.25,0,0,1,0,.354Z"
                    fill="currentColor"
                  />
                </svg>
              </button>
              <header
                class="chakra-modal__header css-epw6ly"
                id="chakra-modal--header-:r3:"
              >
                Add New Customer
              </header>
              <p
                class="chakra-text css-bp306l"
                data-testid="modal-content"
              >
                Lorem Ipsum is simply dummy text of the printing and typesetting industry.
              </p>
            </section>
          </div>
        </div>
        <div
          aria-hidden="true"
          data-aria-hidden="true"
          data-focus-guard="true"
          style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
          tabindex="0"
        />
      </div>
    </div>
  </body>,
  "container": <div
    aria-hidden="true"
    data-aria-hidden="true"
  />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Modal Component Should match snapshoot for Modal with size xl 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div
      aria-hidden="true"
      data-aria-hidden="true"
    />
    <div
      class="chakra-portal"
    >
      <div
        class="chakra-modal__overlay css-13d1tkw"
        style="opacity: 0;"
      >
        <div
          aria-hidden="true"
          data-aria-hidden="true"
          data-focus-guard="true"
          style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
          tabindex="0"
        />
        <div
          data-focus-lock-disabled="false"
        >
          <div
            class="chakra-modal__content-container css-1oxhx2p"
            tabindex="-1"
          >
            <section
              aria-labelledby="chakra-modal--header-:r5:"
              aria-modal="true"
              class="chakra-modal__content css-1bn8s9p"
              id="chakra-modal-:r5:"
              role="dialog"
              style="opacity: 0; transform: scale(0.95) translateZ(0);"
              tabindex="-1"
            >
              <button
                aria-label="Close"
                class="chakra-modal__close-btn css-6sdaxy"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="chakra-icon css-onkibi"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M.439,21.44a1.5,1.5,0,0,0,2.122,2.121L11.823,14.3a.25.25,0,0,1,.354,0l9.262,9.263a1.5,1.5,0,1,0,2.122-2.121L14.3,12.177a.25.25,0,0,1,0-.354l9.263-9.262A1.5,1.5,0,0,0,21.439.44L12.177,9.7a.25.25,0,0,1-.354,0L2.561.44A1.5,1.5,0,0,0,.439,2.561L9.7,11.823a.25.25,0,0,1,0,.354Z"
                    fill="currentColor"
                  />
                </svg>
              </button>
              <header
                class="chakra-modal__header css-epw6ly"
                id="chakra-modal--header-:r5:"
              >
                Add New Customer
              </header>
              <p
                class="chakra-text css-bp306l"
                data-testid="modal-content"
              >
                Lorem Ipsum is simply dummy text of the printing and typesetting industry.
              </p>
            </section>
          </div>
        </div>
        <div
          aria-hidden="true"
          data-aria-hidden="true"
          data-focus-guard="true"
          style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
          tabindex="0"
        />
      </div>
    </div>
  </body>,
  "container": <div
    aria-hidden="true"
    data-aria-hidden="true"
  />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Modal Component Should match snapshoot for Modal with size xs 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div
      aria-hidden="true"
      data-aria-hidden="true"
    />
    <div
      class="chakra-portal"
    >
      <div
        class="chakra-modal__overlay css-13d1tkw"
        style="opacity: 0;"
      >
        <div
          aria-hidden="true"
          data-aria-hidden="true"
          data-focus-guard="true"
          style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
          tabindex="0"
        />
        <div
          data-focus-lock-disabled="false"
        >
          <div
            class="chakra-modal__content-container css-1oxhx2p"
            tabindex="-1"
          >
            <section
              aria-labelledby="chakra-modal--header-:r1:"
              aria-modal="true"
              class="chakra-modal__content css-1bn8s9p"
              id="chakra-modal-:r1:"
              role="dialog"
              style="opacity: 0; transform: scale(0.95) translateZ(0);"
              tabindex="-1"
            >
              <button
                aria-label="Close"
                class="chakra-modal__close-btn css-6sdaxy"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="chakra-icon css-onkibi"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M.439,21.44a1.5,1.5,0,0,0,2.122,2.121L11.823,14.3a.25.25,0,0,1,.354,0l9.262,9.263a1.5,1.5,0,1,0,2.122-2.121L14.3,12.177a.25.25,0,0,1,0-.354l9.263-9.262A1.5,1.5,0,0,0,21.439.44L12.177,9.7a.25.25,0,0,1-.354,0L2.561.44A1.5,1.5,0,0,0,.439,2.561L9.7,11.823a.25.25,0,0,1,0,.354Z"
                    fill="currentColor"
                  />
                </svg>
              </button>
              <header
                class="chakra-modal__header css-epw6ly"
                id="chakra-modal--header-:r1:"
              >
                Add New Customer
              </header>
              <p
                class="chakra-text css-bp306l"
                data-testid="modal-content"
              >
                Lorem Ipsum is simply dummy text of the printing and typesetting industry.
              </p>
            </section>
          </div>
        </div>
        <div
          aria-hidden="true"
          data-aria-hidden="true"
          data-focus-guard="true"
          style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
          tabindex="0"
        />
      </div>
    </div>
  </body>,
  "container": <div
    aria-hidden="true"
    data-aria-hidden="true"
  />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
